libraries/type-length-value/src/state.rs

```diff
@@ -346,25 +453,111 @@ impl<'data> TlvStateMut<'data> {
             if self.data.len() < value_end {
                 return Err(ProgramError::InvalidAccountData);
             }
-            Ok(&mut self.data[value_start..value_end])
+            Ok((&mut self.data[value_start..value_end], entry_number))
         } else {
             Err(TlvError::TypeAlreadyExists.into())
         }
     }
 
-    /// Reallocate the given number of bytes for the given SplDiscriminate. If the new
-    /// length is smaller, it will compact the rest of the buffer and zero out
-    /// the difference at the end. If it's larger, it will move the rest of
-    /// the buffer data and zero out the new data.
+    /// Allocate space for the given type, where no repeating discriminators are
+    /// allowed
+    pub fn alloc_unique<V: SplDiscriminate>(
+        &mut self,
+        length: usize,
+    ) -> Result<&mut [u8], ProgramError> {
+        let (bytes, _) = self.alloc::<V>(length, Some(0))?;
+        Ok(bytes)
+    }
+
+    /// Allocate space for the given type, where repeating discriminators _are_
+    /// allowed
+    pub fn alloc_allow_repeating<V: SplDiscriminate>(
+        &mut self,
+        length: usize,
+    ) -> Result<(&mut [u8], usize), ProgramError> {
+        self.alloc::<V>(length, None)
+    }
+
+    /// Allocates and serializes a new TLV entry from a Pod type
+    fn add_entry<V: SplDiscriminate + Pod>(
+        &mut self,
+        value: &V,
+        entry_number: Option<usize>,
+    ) -> Result<usize, ProgramError> {
+        let (data, entry_number) = self.alloc::<V>(size_of::<V>(), entry_number)?;
+        data.copy_from_slice(bytemuck::bytes_of(value));
+        Ok(entry_number)
+    }
```

Comment:
This seems unnecessary since you already have `init_value_with_repetition`, probably just remove this and use that? It's kind of an antipattern to use zero-copy serde and then copy, so it's probably best to not encourage it

Reply:
That's fair. I was trying to get around having to set each field or set the data in an extra step. But better off removing prob